### PR TITLE
Fixing source task close error + other fixes

### LINKED ISSUE #253: Cosmos errors when closing a source connector
## Description
When deleting a source connector, I encounter the following error messages, most likely coming from the Cosmos SDK.

```
[2021-01-15 17:18:24,171] ERROR Unexpected exception RntbdTransportClient({"id":1,"isClosed":true,"configuration":{"wireLogLevel":null,"connectionEndpointRediscoveryEnabled":false,"maxChannelsPerEndpoint":130,"maxRequestsPerChannel":30,"bufferPageSize":8192,"maxBufferCapacity":8388608,"requestTimeoutInNanos":5000000000,"requestTimerResolutionInNanos":100000000,"shutdownTimeoutInNanos":15000000000,"connectionAcquisitionTimeoutInNanos":5000000000,"connectTimeoutInMillis":5000,"idleConnectionTimeoutInNanos":0,"idleConnectionTimerResolutionInNanos":100000000,"idleEndpointTimeoutInNanos":3600000000000,"receiveHangDetectionTimeInNanos":65000000000,"sendHangDetectionTimeInNanos":10000000000,"maxConcurrentRequestsPerEndpoint":10000,"userAgent":{"suffix":"APN/1.0 Microsoft/1.0 KafkaConnect/null","userAgent":"azsdk-java-cosmos/4.8.0 Linux/5.4.0-1035-azure JRE/11.0.8 APN/1.0 Microsoft/1.0 KafkaConnect/null"}},"serviceEndpoints":{"count":0,"items":[]}}) is closed received while reading from store. (com.azure.cosmos.implementation.directconnectivity.StoreReader)
com.azure.cosmos.implementation.directconnectivity.TransportException: RntbdTransportClient({"id":1,"isClosed":true,"configuration":{"wireLogLevel":null,"connectionEndpointRediscoveryEnabled":false,"maxChannelsPerEndpoint":130,"maxRequestsPerChannel":30,"bufferPageSize":8192,"maxBufferCapacity":8388608,"requestTimeoutInNanos":5000000000,"requestTimerResolutionInNanos":100000000,"shutdownTimeoutInNanos":15000000000,"connectionAcquisitionTimeoutInNanos":5000000000,"connectTimeoutInMillis":5000,"idleConnectionTimeoutInNanos":0,"idleConnectionTimerResolutionInNanos":100000000,"idleEndpointTimeoutInNanos":3600000000000,"receiveHangDetectionTimeInNanos":65000000000,"sendHangDetectionTimeInNanos":10000000000,"maxConcurrentRequestsPerEndpoint":10000,"userAgent":{"suffix":"APN/1.0 Microsoft/1.0 KafkaConnect/null","userAgent":"azsdk-java-cosmos/4.8.0 Linux/5.4.0-1035-azure JRE/11.0.8 APN/1.0 Microsoft/1.0 KafkaConnect/null"}},"serviceEndpoints":{"count":0,"items":[]}}) is closed
[2021-01-15 17:18:24,171] ERROR Unexpected exception RntbdTransportClient({"id":1,"isClosed":true,"configuration":{"wireLogLevel":null,"connectionEndpointRediscoveryEnabled":false,"maxChannelsPerEndpoint":130,"maxRequestsPerChannel":30,"bufferPageSize":8192,"maxBufferCapacity":8388608,"requestTimeoutInNanos":5000000000,"requestTimerResolutionInNanos":100000000,"shutdownTimeoutInNanos":15000000000,"connectionAcquisitionTimeoutInNanos":5000000000,"connectTimeoutInMillis":5000,"idleConnectionTimeoutInNanos":0,"idleConnectionTimerResolutionInNanos":100000000,"idleEndpointTimeoutInNanos":3600000000000,"receiveHangDetectionTimeInNanos":65000000000,"sendHangDetectionTimeInNanos":10000000000,"maxConcurrentRequestsPerEndpoint":10000,"userAgent":{"suffix":"APN/1.0 Microsoft/1.0 KafkaConnect/null","userAgent":"azsdk-java-cosmos/4.8.0 Linux/5.4.0-1035-azure JRE/11.0.8 APN/1.0 Microsoft/1.0 KafkaConnect/null"}},"serviceEndpoints":{"count":0,"items":[]}}) is closed received while reading from store. (com.azure.cosmos.implementation.directconnectivity.StoreReader)
com.azure.cosmos.implementation.directconnectivity.TransportException: RntbdTransportClient({"id":1,"isClosed":true,"configuration":{"wireLogLevel":null,"connectionEndpointRediscoveryEnabled":false,"maxChannelsPerEndpoint":130,"maxRequestsPerChannel":30,"bufferPageSize":8192,"maxBufferCapacity":8388608,"requestTimeoutInNanos":5000000000,"requestTimerResolutionInNanos":100000000,"shutdownTimeoutInNanos":15000000000,"connectionAcquisitionTimeoutInNanos":5000000000,"connectTimeoutInMillis":5000,"idleConnectionTimeoutInNanos":0,"idleConnectionTimerResolutionInNanos":100000000,"idleEndpointTimeoutInNanos":3600000000000,"receiveHangDetectionTimeInNanos":65000000000,"sendHangDetectionTimeInNanos":10000000000,"maxConcurrentRequestsPerEndpoint":10000,"userAgent":{"suffix":"APN/1.0 Microsoft/1.0 KafkaConnect/null","userAgent":"azsdk-java-cosmos/4.8.0 Linux/5.4.0-1035-azure JRE/11.0.8 APN/1.0 Microsoft/1.0 KafkaConnect/null"}},"serviceEndpoints":{"count":0,"items":[]}}) is closed
... (repeats about 100 more times) ...
[2021-01-15 17:18:38,756] WARN Fail the session read dbs/kafkaconnect/colls/kafka-leases/docs/siva-test.documents.azure.com_kafkaconnect_kafka..0, request session token <empty> (com.azure.cosmos.implementation.directconnectivity.ConsistencyReader)
[2021-01-15 17:18:38,756] WARN SessionTokenMismatchRetryPolicy not retrying because it has exceeded the time limit. Retry count = 101 (com.azure.cosmos.implementation.SessionTokenMismatchRetryPolicy)
```

## Expected Behavior

Connector closes out and this stack trace doesn't happen

## Reproduce

1. Create source connector
2. Wait for connector to come up
3. Monitor docker logs for connect container (`docker logs -f connect`)
4. Close source connector and observe the logs

## Additional Context
N/A

### PR DESCRIPTION
**Pull request by sivamu**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
- Fixes errors when source task is closed
- Getting the connector logging working again
- Pom changes to fix Cosmos user agent, shade outputs

## Issues Closed or Referenced

- Related #253

### CODE REVIEW

File: src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java [hunk 1]
```diff
@@ -227,11 +227,6 @@ public void stop() {
             changeFeedProcessor.stop();
             changeFeedProcessor = null;
         }
-
-        if (client != null) {
-            client.close();
-            client = null;
-        }
```
Comment (ryancrawcour) [inline src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java:234]: so we don't have a dispose now to clean up resources that might be open? 
was this what was causing the error? 
now if the connector is stopped and a cosmos client is open, what happens? is it cleaned up properly?

**Review by ryancrawcour — CHANGES_REQUESTED** [review]

File: src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java (hunk 1 above)
Comment (sivamu) [reply src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java:234]: This was strangely causing the error. I modeled the approach based on how [Azure-samples did it](https://github.com/Azure-Samples/azure-cosmos-java-sql-api-samples/blob/main/src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedProcessor.java#L105) where they only closed the Changefeedprocessor. Ideally we do want to clean up all resources and I'll do a bit more testing into recreating the connectors when the clients are up.

Comment (sivamu) [reply src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java:234]: I changed the order of the client close and that didn't make a difference. Tested with deleting and recreating the connector several times and didn't encounter any errors. I suppose we could ask the Cosmos team whether why this behavior is happening.

Comment (ryancrawcour) [reply src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java:234]: yeah that would be a good idea. let's go ahead and merge this, but not close the issue so we can track this issue with the PG.

**Review by ryancrawcour — APPROVED** [review]

**Comment by ryancrawcour** [comment]
merging PR as the change fixes the error we were seeing in Connect when stopping the connector. Need to track issue #253 with the PG to figure out why this is happening.

**Pull request merged**